mobile_verifier/src/boosting_oracles/mod.rs

```diff
@@ -1,184 +1,600 @@
 pub mod assignment;
-
-use crate::{
-    geofence::{Geofence, GeofenceValidator},
-    Settings,
-};
-pub use assignment::{Assignment, HexAssignments};
+pub mod footfall;
+pub mod landtype;
+pub mod urbanization;
+
+use std::path::Path;
+use std::pin::pin;
+use std::sync::Arc;
+use std::{collections::HashMap, path::PathBuf};
+
+use crate::boosting_oracles::assignment::HexAssignments;
+use crate::coverage::SignalLevel;
+pub use assignment::Assignment;
+use chrono::{DateTime, Duration, Utc};
+use file_store::{file_sink::FileSinkClient, traits::TimestampEncode, FileStore};
+use futures_util::{Stream, StreamExt, TryFutureExt, TryStreamExt};
+use helium_proto::services::poc_mobile as proto;
 use hextree::disktree::DiskTreeMap;
+use rust_decimal::prelude::ToPrimitive;
+use rust_decimal_macros::dec;
+use sqlx::{FromRow, PgPool, QueryBuilder};
+use task_manager::ManagedTask;
+use tokio::{fs::File, io::AsyncWriteExt, sync::Mutex};
+pub use urbanization::Urbanization;
+use uuid::Uuid;
+
+pub trait DataSet: HexAssignment + Send + Sync + 'static {
+    const TYPE: DataSetType;
+
+    fn timestamp(&self) -> Option<DateTime<Utc>>;
 
-pub trait BoostedHexAssignments: Send + Sync {
-    fn assignments(&self, cell: hextree::Cell) -> anyhow::Result<HexAssignments>;
+    fn update(&mut self, path: &Path, time_to_use: DateTime<Utc>) -> anyhow::Result<()>;
+
+    fn is_ready(&self) -> bool;
 }
 
-pub struct HexBoostData {
-    urbanized: DiskTreeMap,
-    usa_geofence: Geofence,
-    footfall: DiskTreeMap,
-    landtype: DiskTreeMap,
+pub struct DataSetDownloaderDaemon<T, A, B, C> {
```

Comment:
this code would be more readable if the DataSetDownloaderDaemon struct and its impls were in their own module.  Likewise the DB functions would also be pushed to its own module

Reply:
the DB functions are in their own module

Reply:
Yeah i meant to say push that mod out to its own file

Reply:
and I have move the downloader  to a separate module as well